tests/SDAccel/hls_with_cppkernel_opencl_kernel_ap_vector_add_args.cpp

```diff
@@ -0,0 +1,88 @@
+/* Simple OpenCL vector addition using SYCL in OpenCL interoperability
+   mode Boost.Compute C++ host API and precompiled kernel.
+
+   Compared to using plain Boost.Compute OpenCL host API, in SYCL the
+   buffer transfers between host and device are implicits with the
+   accessors.
+*/
+
+/* REQUIRES: xilinx-xocc
+   RUN: %{execute}%s | %{filecheck} %s
+   CHECK: Result:
+   CHECK-NEXT: 6 8 11
+*/
+#include <iostream>
+#include <iterator>
+#include <boost/compute.hpp>
+#include <boost/test/minimal.hpp>
+
+#include <CL/sycl.hpp>
+
+//#include "cnl/FPGA/Xilinx/include/ap_int.h"
+//#include "/proj/xbuilds/2017.2_sdx_qualified_latest/installs/lin64/SDx/2017.2/Vivado_HLS/include/ap_int.h"
+#include "ap_int.h"
+
+using namespace cl::sycl;
+
+constexpr size_t N = 3;
+using Vector = ap_int<256>[N];
```

Comment:
Great use case!
Of course you have to explain what it is about... :-)

Reply:
I was thinking that it would be even cleaner to have
```
using Vector = std::array<ap_int<256>, N>;
```
and add
```
#include <array>
```
above.